src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -149,19 +162,116 @@ public int getTotalMessageCount(){
     return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
   
-  /* Returns all laptops */
-  public List<Laptop> getAllLaptops(){
-	 List<Laptop> laptops = new ArrayList<>();
-	 laptops.add(new Laptop("Dell","Black","Windows",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",13,1200.00));
-	 laptops.add(new Laptop("Dell","Grey","Windows",15,800.00));
-	 laptops.add(new Laptop("Dell","Black","Linux",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",15,1699.00));
-	 laptops.add(new Laptop("Macbook Pro","Black","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Black","Mac",15,1900.00));
-	 laptops.add(new Laptop("Macbook Pro","Grey","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Grey","Mac",15,1900.00));
-
-	 return laptops;
-  }
-}
\ No newline at end of file
+  /**
+   * @return list of all Laptops
+   */
+  public List<Laptop> getAlllaptopList(){
```

Comment:
One thing to consider is that this returns the "live" list, so callers of this function can make changes to the underlying data. That might be fine, but it would allow callers to delete laptops for example. One solution to this would be to return a copy of the list, instead of the actual list. It's probably not worth worrying about if this is temporary code that's going to be changed, but it's something to think about.

Reply:
I will ask you for more info on this. I thought that the list I am returning is only being generated when the function is called, so isn't it just the list being returned the only live list? (not the one that has all the laptops in it)

Reply:
The List is created in the constructor, and then lives in memory. You use this List for the rest of the functions in this class. So when you return the List from this function, you're potentially giving callers of this function access to the underlying data. For example I could call this function and then change all of the Laptops to Windows operating systems, which would then have side effects for anybody else using this class.

Like I said I don't think you should worry about it too much, especially if this code is going to change in the near future anyway. Just something to think about.